src/main/java/org/testng/IAlterTestName.java

```diff
@@ -0,0 +1,19 @@
+package org.testng;
+
+/**
+ * This interface lets you alter the test name of an {@link ITestResult} object.
+ * Sample : <br>
+ *
+ * <pre>
+ *     if (testResult instanceOf IAlterTestName) {
+ *         ((ITestResult) testResult).setTestName(newName);
+ *     }
+ *
+ * </pre>
+ */
+public interface IAlterTestName {
```

Comment:
I don't know if we need a new interface here: we are just adding a new method.
@cbeust What do you think?